share/doc/DEVELOPERS_GUIDE.rst

```diff
@@ -494,29 +494,39 @@ To update the state of a submodule the following commands can be used:
-Once this pull request is merged, another pull request from stable to master is created in the submodule automatically.
-And as for the master branch (see previous section) and pull request in :code:`votca/votca` is opened automatically to
-update the submodules in the stable branch as well.
-So ultimately there will 4 automatically created pull requests, (a) in the submodule from stable to master, (b) in the main repository
-to update the stable submodules, (c) to update the master submodules in the main repository once (a) is merged and (d) an update from stable to master in the main repository once (b) is merged. From time to time (a) will need some manual intervention to resolve conflicts. To minimize the manual work on (d), it is usually best to merge the pull requests in the order (a), (c), (b).
-The background is that when (d) gets created by merging (b) and the submodules in master are already up to date from merging (c) there will not be any merge conflicts. 
+The :code:`stable` branch contains the latest release with the most uptodate bug fixes since the release.
+Only in very limited circumstances should new features be merged into the :code:`stable` branch.
+Developers can add bug fixes by making a pull request with the :code:`stable` branch as target.
 
+As the :code:`master` branch of each repo is a child of each repo's :code:`stable` branch,  
+any bugfix added to a repos :code:`stable` branch will also need to be pulled into its :code:`master` branch. If the bugfix
+is added in one of the child repositories (not :code:`votca/votca`) then :code:`votca/votca` will also need to
+reflect these changes.
+
+Keeping the repositories synchronised can be difficult. In order to help keep the :code:`master` branches and :code:`votca/votca`
+synchronised with changes in the :code:`stable` branch of a child repository the generation of four pull requests are
+automatically generated anytime a bugfix is made to the :code:`stable` branch of a child repository.
+
+E.g. if :code:`hot-bug-fix` is merged into the :code:`stable` branch of :code:`tools`:
+
+1. A pull request is created to merge :code:`stable` from :code:`tools` (child repo) into :code:`master` of :code:`tools` (child repo).
+2. A pull request is created to merge :code:`stable` from :code:`tools` (child repo) into :code:`stable` of :code:`votca/votca` (parent repo). This shoud consiste of updating the submodules in the :code:`stable` branch of votca/votca.
+3. A pull request is created to merge :code:`master` from :code:`tools` (child repo) into :code:`master` of :code:`votca/votca` (parent repo). Again this should consist of updating the submodules but in the :code:`master` branch of votca/votca.
+4. Finally, a pull request is made from :code:`stable` from :code:`votca/votca` (parent repo) to :code:`master` of :code:`votca/votca` (parent repo).
+
+To minimize manual work, it is usually best to merge the pull requests in the order that hey have been shown in the example. 
```

Comment:
This is better now, but a couple of things are still not quite right.(3) is only created after (1) is merged. And (4) is only created (2) is merged.

Reply:
@JensWehner

Reply:
yes?

Reply:
Do you understand it?